### DISCUSSION TITLE
How to use regex expressions with Postgres?

### DISCUSSION THREAD

**Discussion by csotiriou** [Blazebit/blaze-persistence#1752] [General]
I'm trying to figure out how to use regex supported by Postgres. I want to have something like: `select * from address where name_el ~ 'ath'`

If this was JPA, I would be able to do something like this like this, since the JPA offers raw SQL support in the argyments:
```
subquery
  .select(subqueryRoot.get("id"))
  .where(cb.sql("name_el like '%ala%'", Address.class).as(Boolean.class));
```

Howeve, when using Blaze, the below methodology
```
var cb = criteriaBuilderFactory.create(em, Address.class);
var criteria = cb
        .page(0, 10)
        .whereExpression("nameEn ~ 'foo'");
```

gives me the error: `Could not parse expression 'nameEn ~ 'foo'',`
The `~` operator exists in Postgres.

Is there something I can do to cover my case?

**Comment by beikov** [comment]
This questions seems very similar to your [other one](https://github.com/Blazebit/blaze-persistence/discussions/1753) for which I just added an answer.

If you want to use a `like` predicate, you can easily do that this way:

```java
var cb = criteriaBuilderFactory.create(em, Address.class);
var criteria = cb
        .page(0, 10)
        .whereExpression("nameEn like 'foo'");
```

or

```java
var cb = criteriaBuilderFactory.create(em, Address.class);
var criteria = cb
        .page(0, 10)
        .where("nameEn").like().value("%foo%").noEscape();
```
